### PR TITLE
[debugger] Merge debugger-libs/Mono.Debugger.Soft

### PR DESCRIPTION
**Pull request by thaystg**
Trying to make mono/debugger-libs/Mono.Debugger.Soft be the same as mono/mono/mcs/class/Mono.Debugger.Soft

Relates to https://github.com/mono/debugger-libs/pull/323

### CODE REVIEW

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/FieldInfoMirror.cs [hunk 1]
```diff
@@ -2,10 +2,8 @@
 using System.Collections.Generic;
 using System.Text;
 using System.Reflection;
-
-#if ENABLE_CECIL
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/FieldInfoMirror.cs:6]: The ENABLE_CECIL stuff was added by:
https://github.com/mono/mono/commit/83d63bf1818d0f151f5ab06a61422caab445f4a7
Maybe its still needed ?

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/DelayedLambdaType.cs [hunk 2]
```diff
@@ -0,0 +1,170 @@
+﻿﻿using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Mono.Debugger.Soft
+{
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/DelayedLambdaType.cs:6]: Why are these types here ? They are not used by the rest of the assembly.

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/ArrayMirror.cs [hunk 3]
```diff
@@ -92,6 +92,14 @@ public class ArrayMirror : ObjectMirror, IEnumerable {
 			vm.conn.Array_SetValues (id, index, vm.EncodeValues (values));
 		}
 
+		public void SetByteValues (byte [] bytes)
+		{
+			if (bytes != null && bytes.Length != Length) {
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/ArrayMirror.cs:97]: This might need a protocol version check.

Comment (thaystg) [reply mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/ArrayMirror.cs:97]: This implementation uses this CMD_ARRAY_REF_SET_VALUES that exists since you write it :), I don't think that we need a protocol version check.

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/FieldInfoMirror.cs (hunk 1 above)
Comment (thaystg) [reply mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/FieldInfoMirror.cs:6]: Yes, I talked to @jbevain and it's needed. Thanks

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/DelayedLambdaType.cs (hunk 2 above)
Comment (thaystg) [reply mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/DelayedLambdaType.cs]: Removed

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/MethodBodyMirror.cs [hunk 4]
```diff
@@ -8,8 +8,7 @@
 
 #if ENABLE_CECIL
 using Mono.Cecil.Cil;
-#else
-using System.Reflection.Emit;
+using Mono.Cecil.Metadata;
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/MethodBodyMirror.cs:11]: Why are these changes needed ? Weren't they in mono's version already ?

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins commit apidiff
